runtime/bundles/org.eclipse.core.jobs/src/org/eclipse/core/internal/jobs/InternalJob.java

```diff
@@ -155,7 +155,7 @@ public abstract class InternalJob extends PlatformObject implements Comparable<I
 	/**
 	 * This signal is used to synchronize Job listener notification
 	 */
-	volatile boolean waitForNotificationFinished;
+	volatile Thread notifyDoneThread;
```

Comment:
I would rename it to "notifyPendingThread", to indicate that notification is not yet done.

Reply:
ok